### DISCUSSION TITLE
Baremetal simulations on RISC-V

### DISCUSSION THREAD

**Discussion by akarxxx1030** [gem5/gem5#1144] [Q&A]
Hey, everyone. I want to run baremetal programs on various RISC-V configurations while not involving the overhead of the OS. I understand gem5 is a full system simulator and that the two modes prevalent somewhat involve the overhead of the Linux OS. (as stated in https://stackoverflow.com/questions/66035699/bare-metal-in-gem5-for-risc-v).

Could anyone provide any insight on whether this is possible? and if so, how? Correct me if any of my points are incorrect.

**Comment by robhau** [comment]
I suggest you to take a look in https://github.com/orgs/gem5/discussions/1132

**Reply by flyeagle0** [reply]
I think the following link is more helpful https://github.com/orgs/gem5/discussions/897

**Reply by robhau** [reply]
Thank you 😄. I was not aware of this discussion / config file option.

**Reply by akarxxx1030** [reply]
Thanks! I will take a look.
